libyara/modules/pe.c

```diff
@@ -1052,7 +1052,7 @@ int pe_valid_dll_name(
     }
   }
 
-  return (l > 0 && l < n);
+  return (l > 0 && l <= n);
```

Comment:
I think this introduces a potential buffer overrun in line 1117, where yr_strdup is called with dll_name as its argument. If dll_name is at the end of the file and without a null termination, pe_valid_dll_name will return true and yr_strdup can be invoked with a non-null-terminated string.